src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy

```diff
@@ -164,39 +167,57 @@ abstract class WorkflowUsingMergedBams extends Workflow {
      * @return
      */
     boolean checkInitialFiles(ExecutionContext context, BasicBamFile[] initialBamFiles) {
+        // Check size
         if (!initialBamFiles) {
-            context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Did not find any BAM files."))
+            context.addError(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Did not find any bam files."))
             return false
         }
 
-        if (isNoControlWorkflow()) {
-            boolean foundAll = true
-            initialBamFiles.each { BasicBamFile it -> foundAll &= it && ((COFileStageSettings) it.getFileStage()).sample.sampleType == Sample.SampleType.TUMOR }
+        if (isControlWorkflow() && initialBamFiles.size()<2) {
+            context.addError(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Bam file list is too small, need one bam for no control and two bams for control workflows."))
+            return false
+        }
 
-            if (!foundAll) {
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Not all found files are tumor BAM files."))
-                return false
-            }
+        // Check, if the bams were created properly
+        if (initialBamFiles.any { it == null }) {
+            context.addError(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("The list of bam files contains an empty entry."))
+            return false
+        }
 
-        } else {
-            boolean foundAll = true
-            BasicBamFile bamControlMerged = initialBamFiles[0]
+        if (initialBamFiles.any { BasicBamFile bam -> bam.fileStage == null }) {
+            context.addError(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("The list of bam files contains an entry with unset sample."))
+            return false
+        }
 
-            for (int i = 1; i < initialBamFiles.size(); i++) {
-                if (initialBamFiles[i] == null || !((COFileStageSettings) initialBamFiles[i].getFileStage()).sample.sampleType == Sample.SampleType.TUMOR) {
-                    context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Tumor BAM is missing."))
-                    foundAll = false
-                }
-            }
+        // Check, if samples are ok (one control, several tumors)
```

Comment:
You make a comment what you do (although this is pretty simple). Write a function with a descriptive name instead. E.g. `checkBamListComposition` or so.